internal/api/dispatch/impl.go

```diff
@@ -106,3 +107,40 @@ func (this *dispatchManager) ProcessRun(ctx context.Context, account string, ser
 	instrumentation.RunCreated(ctx, run.Recipient, entity.ID, run.Url, entity.Service, protocol.GetLabel())
 	return entity.ID, correlationID, nil
 }
+
+func (this *dispatchManager) ProcessCancel(ctx context.Context, account string, cancel generic.CancelInput) (runID, correlationID uuid.UUID, err error) {
+	var run db.Run
+
+	protocol := *protocols.SatelliteProtocol
+
+	if err := this.db.First(&run, cancel.RunId).Error; err != nil {
+		instrumentation.PlaybookRunCancelError(ctx, err, protocol.GetLabel())
+		return uuid.UUID{}, run.CorrelationID, &RunNotFoundError{err: err, runID: *cancel.RunId}
+	}
+
+	signalMetadata := protocol.BuildCancelMetadata(cancel, run.CorrelationID, this.config)
```

Comment:
Before proceeding we should check what status the run has. If it's no longer running then there's no point in sending the cancelation message.